### ISSUE TITLE
volume.linear_interpolation does not interpolate any pixel

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/xdem#84]
volume.linear_interpolation does not work as expected.
It is because of this line https://github.com/GlacioHack/xdem/blob/4e5ac649fec4acac88f2754f684ec22d941eb8c1/xdem/volume.py#L188
There, the mask get values 254 and 255, not 0 as expected by rasterio's fillnodata. Hence, no pixel is ever interpolated.

**Comment by erikmannerfelt** [comment]
I don't quite get this description. Can you give a minimal example? I tried casting types below:

```python
In [1]: import numpy as np

In [2]: arr = np.random.randint(0, 2, size=(4, 4)).astype(bool)

In [3]: arr
Out[3]:
array([[False, False, False,  True],
       [ True, False, False, False],
       [ True, False, False, False],
       [ True, False, False, False]])

In [4]: arr.astype("uint8")
Out [4]:
array([[0, 0, 0, 1],
       [1, 0, 0, 0],
       [1, 0, 0, 0],
       [1, 0, 0, 0]], dtype=uint8)
```
`True` is cast as `1` and `False` is 0. I don't  get where `244` comes into the picture.

**Comment by erikmannerfelt** [comment]
Looking at the [docs](https://rasterio.readthedocs.io/en/latest/api/rasterio.fill.html#rasterio.fill.fillnodata), any value that is not `0` should be parsed as an inlier. Therefore `~nan_mask.astype("uint8")` should work. Maybe it has to do with parentheses? `~(nan_mask.astype("uint8"))` is maybe what it should say?

**Comment by erikmannerfelt** [comment]
I found the issue!

```python
In [5]: ~arr.astype("uint8")
Out [5]: 
array([[255, 255, 255, 254],
       [254, 255, 255, 255],
       [254, 255, 255, 255],
       [254, 255, 255, 255]], dtype=uint8)

In [6]:  (~arr).astype("uint8")
Out [6]:
array([[1, 1, 1, 0],
       [0, 1, 1, 1],
       [0, 1, 1, 1],
       [0, 1, 1, 1]], dtype=uint8)
```
The problem comes from the type conversion being done before the inversion. "bitwise not 255" is weirdly enough 244!

**Comment by adehecq** [comment]
Solved by #83
